affine/__init__.py

```diff
@@ -286,6 +286,62 @@ def determinant(self):
         return a * e - b * d
 
     @property
+    def _scaling(self):
+        """The absolute scaling factors of the transformation.
+
+        This tuple represents the absolute value of the scaling factors of the
+        transformation, sorted from bigger to smaller.
+        """
+        a, b, _, d, e, _, _, _, _ = self
+
+        # The singular values are the square root of the eigenvalues
+        # of the matrix times its transpose, M M*
+        # Computing trace and determinant of M M*
+        trace = a**2 + b**2 + d**2 + e**2
+        det = (a * e - b * d)**2
+
+        delta = trace**2 / 4 - det
+        if delta < 1e-12:
+            delta = 0
+
+        l1 = math.sqrt(trace / 2 + math.sqrt(delta))
+        l2 = math.sqrt(trace / 2 - math.sqrt(delta))
+        return l1, l2
+
+    @property
+    def eccentricity(self):
+        """The eccentricity of the affine transformation.
+
+        This value represents the eccentricity of an ellipse under
+        this affine transformation.
+
+        Raises NotImplementedError for improper transformations.
+        """
+        if self.is_proper or self.is_degenerate:
+            l1, l2 = self._scaling
+            return math.sqrt(l1 ** 2 - l2 ** 2) / l1
+        else:
+            raise NotImplementedError
+
+    @property
+    def rotation_angle(self):
+        """The rotation angle in degrees of the affine transformation.
+
+        This is the rotation angle in degrees of the affine transformation,
+        assuming it is in the form M = R S, where R is a rotation and S is a
+        scaling.
+
+        Raises NotImplementedError for improper transformations.
+        """
+        a, b, _, c, d, _, _, _, _ = self
+        if self.is_proper or self.is_degenerate:
+            l1, _ = self._scaling
+            y, x = c / l1, a / l1
+            return math.atan2(y, x) * 180 / math.pi
+        else:
+            raise NotImplementedError
```

Comment:
What would you think about a unique custom error like an `UndefinedScalingError` that derived from a root `AffineError(Exception)` instead? I'm thinking of Brett Slakin's advice in "Effective Python" about helping users understand whether they're using Python wrong or finding Affine bugs/limitations.